pkgs/servers/search/sonic-server/default.nix

```diff
@@ -30,6 +30,11 @@ rustPlatform.buildRustPackage rec {
     })
   ];
 
+  # Work around https://github.com/NixOS/nixpkgs/issues/166205.
+  env = lib.optionalAttrs stdenv.cc.isClang {
+    NIX_LDFLAGS = "-l${stdenv.cc.libcxx.cxxabi.libName}";
+  };
```

Comment:
stdenv.cc.isClang can be true while stdenv.cc.libxx is null so i've been using `(stdenv.cc.libcxx != null)` as the check.
```console
nix-repl> builtins.currentSystem
"x86_64-linux"
nix-repl> clang16Stdenv.cc.libcxx
null
nix-repl> clang16Stdenv.cc.isClang
true
```

also `NIX_LDFLAGS` will link `-lc++abi` into everything it builds, there seem to be various things getting built by cargo in this package so this could be undesirable.